allennlp/semparse/type_declarations/type_declaration.py

```diff
@@ -0,0 +1,603 @@
+"""
+This module defines some classes that are generally useful for defining a type system for a new
+domain. We inherit the type logic in ``nltk.sem.logic`` and add some functionality on top of it
+here. There are two main improvements:
+1) Firstly, we allow defining multiple basic types with their own names (see ``NamedBasicType``).
+2) Secondly, we allow defining function types that have placeholders in them (see
+``PlaceholderType``).
+We also extend NLTK's ``LogicParser`` to define a ``DynamicTypeLogicParser`` that knows how to deal
+with the two improvements above.
+"""
+from typing import Dict, List, Optional, Set, Tuple, Union
+from collections import defaultdict
+
+from overrides import overrides
+from nltk.sem.logic import Expression, ApplicationExpression, ConstantExpression, LogicParser, Variable
+from nltk.sem.logic import Type, BasicType, ComplexType as NltkComplexType, ANY_TYPE
+
+START_SYMBOL = '@START@'
```

Comment:
We already have a start symbol here, can you use this:

https://github.com/allenai/allennlp/blob/master/allennlp/data/dataset_readers/seq2seq.py#L17

Reply:
I don't much like the idea of importing that symbol in other places.  We could have a symbol in `common`, or something, that gets imported in both places.  Yes?

Reply:
Sure - I just don't want there to be two global constants with the same name defined in different places